server/src/main/java/org/elasticsearch/index/mapper/NestedPathFieldMapper.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.lucene.Lucene;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.query.QueryShardContext;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class NestedPathFieldMapper extends MetadataFieldMapper {
+
+    public static final String NAME_PRE_V8 = "_type";
+    public static final String NAME = "_nested_path";
+
+    public static String name(Settings settings) {
+        if (Version.indexCreated(settings).before(Version.V_8_0_0)) {
+            return NAME_PRE_V8;
+        }
+        return NAME;
+    }
+
+    public static Query filter(Settings settings, String path) {
+        return new TermQuery(new Term(name(settings), new BytesRef(path)));
+    }
+
+    public static Field field(Settings settings, String path) {
+        return new Field(name(settings), path, Defaults.FIELD_TYPE);
+    }
+
+    public static class Defaults {
+
+        public static final MappedFieldType FIELD_TYPE = new NestedPathFieldType();
+
+        static {
+            FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
```

Comment:
Should it bee `IndexOptions.NONE` ?

Reply:
We still need to search against it, so we need to store matching docs, unless I'm misunderstanding something?

Reply:
you're right, I mixed this with doc values, sorry for the noise.